projects/pastanaga-angular/src/lib/toast/toast.component.ts

```diff
@@ -38,14 +58,29 @@ export class ToastComponent implements OnInit {
     _action?: () => any;
     _class = '';
 
-    constructor() {}
+    constructor(private rendererFactory: RendererFactory2) {
+        this.renderer = rendererFactory.createRenderer(null, null);
+    }
 
     ngOnInit(): void {
         this.setupAutoClause();
     }
 
+    ngAfterViewInit() {
+        this.addClass(`${this._class}-wrapper`, this.toastWrapper);
+    }
+
     setupAutoClause() {
-        setTimeout(() => this.dismiss.emit(this._id), DEFAULT_DELAY);
+        const DELAY = !this._actionButtonLabel ? TOAST_DEFAULT : TOAST_BUTTON;
+        timer(DELAY)
+            .pipe(
+                take(1),
+                tap(() => {
+                    this.addClass('pa-toast-animate-out', this.toastWrapper);
+                }),
+                switchMap(() => timer(TOAST_ANIMATE_OUT))
+            )
+            .subscribe(() => this.dismiss.emit(this._id));
```

Comment:
@julienCarret helped me with this (and a lot of other stuff!) vs. using nested timeouts